### ISSUE TITLE
Choose 8 as opposed to default of 4-connectivity in measure.regionprops perimeter

### ISSUE THREAD

**Issue by alexpawlowski** [scikit-image/scikit-image#2801]
I realize the trade-offs between additional functionality at the expense of maintainability / good scientific practice, but is there a way to allow a user to select the `neighbourhood` when calculating perimeter to use `8` rather than just the default `4`? 

I have some small labeled regions with areas smaller than 20 pixels that are returning perimeters that would correlate with a circularity over 1. 

I might remove them from my data, but I would be interested in if increasing my neighbourhood would help me capture smaller regions of interest (at expense of recording what should be dropped). 

https://github.com/scikit-image/scikit-image/blob/71903adbe1bd9866d4798582eeaf85a4d28db032/skimage/measure/_regionprops.py#L269

**Comment by stefanv** [comment]
I don't have strong feelings about this one; I'd be happy with 8 being the default too.  At this point, however, it would be hard to implement that without breaking existing code.  We could always add `.perimeter8` or `perimeter_diag`?

**Comment by jni** [comment]
I would very much like this to be optional somehow. Note that I generally prefer the SciPy term of "connectivity", ie 4-neighbours = 1-connected, 8-neighbours = 2-connected. In 3D, it goes (6, 1), (18, 2), (26, 3). I find it much more intuitive to generalise. (Can you list the 4D equivalents off the top of your head?)

In this scenario, the new property becomes the rather lovely `perimeter2`. ;)

**Comment by alexpawlowski** [comment]
Thanks for answering so quickly! 

I think the options to provide this as an optional feature rather than the default is a good place to be. I am not as well versed in the technical reasons for connectivity, but I feel safe in assuming that 4 neighbours / 1-connected was chosen for a decent reason. So an enhancement that adds this feature as another key to play with would be awesome. 

I'm using this package for a paper - my first real dive into image segmentation - so glad to add another citation for this great package.

**Comment by stefanv** [comment]
@jni Agreed, that's a better naming scheme.  @alexpawlowski Thanks for highlighting this issue!  Would you be interested in making a PR to address it?

I hope you're having a good experience with skimage; please let us know if there's anything we can do improve it.

**Comment by alexpawlowski** [comment]
Sure @stefanv I will submit a PR to address. Thanks!
